app/services/catalog/discard_restore.rb

```diff
@@ -0,0 +1,47 @@
+module Catalog
+  module DiscardRestore
+    extend ActiveSupport::Concern
+
+    CHILD_DISCARD_TIME_LIMIT = 30
+
+    module ClassMethods
+      def destroy_dependencies(*klass_names)
+        klass_names.each do |klass|
+          define_method "discard_#{klass}" do
+            collection = send(klass)
+
+            if collection.map(&:discard).any? { |result| result == false }
```

Comment:
What if `discard!` was used here? Would allow the removal of the checking logic.
https://github.com/jhawthorn/discard#discard-a-record

Reply:
Hmm, that's interesting, though then we do end up losing some of our logging here, right? We would need to tap into the error and re-raise anyway, I think, if we wanted to preserve these log messages. If we don't care, then sure, I'm good with changing to use `collection.map(&:discard!)`

Reply:
Good point about the log messages.  We could still manage them by rescuing / reraising our own with the logging, but that seems wrong and in the end additional logic that I was trying to mitigate.